spec/models/assignment_spec.rb

```diff
@@ -1026,7 +1025,12 @@
           result = s.get_latest_result
           result.total_mark = total_mark
           result.marking_state = Result::MARKING_STATES[:complete]
-          result.save
+          @assignment.rubric_criteria.each do |cri|
+            result.marks.create!(markable_id: cri.id,
+                                 markable_type: RubricCriterion,
+                                 mark: (total_mark * 4.0 / 20).round)
```

Comment:
The rounding invalidates the meaning of the `total_marks` array. On the other hand, the array could only be consistent with real marks if weights were assigned to the criteria. @aclar48, if you could clean up this test and create another one for flexible criteria, that would be much appreciated.